src/client/splitpane.ts

```diff
@@ -0,0 +1,140 @@
+    // for each command, register handlers that can send requests to server
+    Object.keys(registeredCommandsAndArgs).forEach(key => {
+      commands.registerCommand(key, () => generateCode(client, registeredCommandsAndArgs[key]));
+    });
+    return null;
+  }
+
+  function generateCode(client: LanguageClient, args: Array<string>) {
+    const currFile: string = window.activeTextEditor.document.fileName;
+    if (!(currFile.endsWith('.json') || currFile.endsWith('.md'))) {
+      window.showErrorMessage(currFile + ' is not a valid OpenAPI specification file format.');
+      return;
+    }
+    let autorestArgs = {};
+    autorestArgs['inputFile'] = window.activeTextEditor.document.fileName;
+    autorestArgs['additionalConfig'] = {};
+    if (args) {
+      args.forEach(element => autorestArgs['additionalConfig'][element] = true);
+    }
+
+    client.sendRequest('autorest.generateCode', autorestArgs).then(result => displayGeneratedCode(<string>result, args));
+  }
+
+  async function writeFilesToDisk(resultObj: any) {
+    let opDirPath: string = path.normalize(resultObj['outputFolder']);
+    if (opDirPath.startsWith(opDirPath)) {
+      opDirPath = opDirPath.replace('file:' + path.sep, '');
+    }
+    deleteFolderRecursive(opDirPath);
+
+    const generatedFiles = resultObj['generatedFiles'];
+    await Object.keys(generatedFiles).forEach(async key => {
+      let filePath = path.normalize(key);
+      if (filePath.startsWith('file:')) {
+        filePath = filePath.replace('file:' + path.sep, '');
+      }
+      let dirName: string = path.dirname(filePath);
+      while (!fs.existsSync(dirName)) {
+        await fs.mkdir(dirName);
+        dirName = path.dirname(dirName);
+      }
+      await writeToDisk(filePath, generatedFiles[key]);
+    });
+    window.showInformationMessage('Code generated under directory ' + opDirPath);
+  }
+
+  async function deleteFolderRecursive(dirPath: string) {
+    if (await fs.exists(dirPath)) {
+      await fs.readdir(dirPath).forEach(async function (file, index) {
+        var curPath = path.join(dirPath, file);
+        if (await fs.lstat(curPath).isDirectory()) { // recurse
+          deleteFolderRecursive(curPath);
+        } else { // delete file
+          await fs.unlink(curPath);
+        }
+      });
+      await fs.rmdir(dirPath);
+    }
+  };
+
+  async function writeContentToMarkdown(resultObj: any, args: Array<string>) {
+    // dump generated code received into a temp file and show that file on the split pane!!!!
+    // this fname is going to basically be the autorest output
+    let opfname = 'output.md';
+    if (args) {
+      opfname = args.join('.') + '.' + opfname;
+    }
+
+    const filePath: string = path.normalize(path.join(os.tmpdir(), opfname));
+    let content: string = '## Generated Code\n';
+    const generatedFiles = resultObj['generatedFiles'];
+
+    Object.keys(generatedFiles).forEach(filePath => {
+      content += '### File Name:'
```

Comment:
why not just use the filename as the heading to the file content? E.g.

## [`Models/Cowbell.cs`](%TMP%\Models\Cowbell.cs)

``` C#
// stuff
class Cowbell
{
    // stuff
}
```

Reply:
which also reminds me, why not derive the highlighting language from the file extension? 🙂 very low hanging fruit, boosts readability like crazy

Reply:
Makrdown allows, language specific highlighting? No idea about it!